### PR TITLE
transaction: Delete removed groups from system state

### PR DESCRIPTION
**Pull request by m-blaha**
Fixes inconsistency in the system state after group removal.

### CODE REVIEW

File: libdnf/base/transaction.cpp [hunk 1]
```diff
@@ -700,6 +700,9 @@ Transaction::TransactionRunResult Transaction::Impl::_run(
                 system_state.set_group_state(group.get_groupid(), state);
                 // save the current xml group definition
                 group.serialize(comps_xml_dir / (group.get_groupid() + ".xml"));
```
Comment (kontura) [inline libdnf/base/transaction.cpp:702]: I am wondering if we want to keep the serialized group xml.
It would probably be cleaner to also delete it?

Comment (m-blaha) [reply libdnf/base/transaction.cpp:702]: Good point.

Comment (m-blaha) [reply libdnf/base/transaction.cpp]: I've added code for deleting the file.

**Review by kontura — APPROVED** [review]
Thanks!

**Pull request merged**